### PR TITLE
✨ added xy ratio calculation tool

### PR DESCRIPTION
**Pull request by dmiller15**
## Description

This PR adds XY ratio stat calculation to the alignment workflow.

Closes https://github.com/d3b-center/bixu-tracker/issues/932

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [x] Tool and workflow passes cwltool validation
- [x] Test of tool with run turned off: https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-reference-pipeline/tasks/bd83b354-9b2f-4e4d-ae8c-8e9f3596b039/
- [x] Test of tool with run turned on: https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-reference-pipeline/tasks/42781d21-d65f-46ed-975f-b873be69f181/#
- [x] Test of new alignment workflow: https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-reference-pipeline/tasks/ebf74a14-08e1-49d1-b467-3b223b4544fc/#

**Test Configuration**:
* Environment: cwltool 3.0.20200807132242, Cavatica
* Test files: See Cavatica tests

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have committed any related changes to the PR

### CODE REVIEW

**Review by migbro — COMMENTED** [review]
Looks good so far!

File: workflows/kfdrc_alignment_wf.cwl [hunk 1]
```diff
@@ -443,6 +443,8 @@ outputs:
   artifact_pre_adapter_summary_metrics: {type: 'File[]?', outputSource: picard_collectsequencingartifactmetrics/pre_adapter_summary_metrics}
   qual_metrics: {type: 'File[]?', outputSource: picard_qualityscoredistribution/metrics}
   qual_chart: {type: 'File[]?', outputSource: picard_qualityscoredistribution/chart}
+  idxstats: {type: 'File', outputSource: samtools_idxstats_xy_ratio/output}
+  xy_ratio: {type: 'File', outputSource: samtools_idxstats_xy_ratio/ratio}
```
Comment (yuankunzhu) [inline workflows/kfdrc_alignment_wf.cwl:447]: for those 2 new files, we probably should put some simple descriptions as `doc`, specially for that `{task-id}.ratio.txt` file. 

also per slack discussion, we probably start adding descriptions to important output files in our prd workflows as most of them are lacking the `doc` part.

**Comment by yuankunzhu** [comment]
cc @zhangb1 for this PR as well. although this is still under reviewing, but we are planning to have `{task_id}.ratio.txt` as one of the alignment workflow, so you can use that to predict gender

File: workflows/kfdrc_alignment_wf.cwl (hunk 1 above)
Comment (dmiller15) [reply workflows/kfdrc_alignment_wf.cwl]: Ok added.

**Comment by dmiller15** [comment]
@yuankunzhu @migbro Took the suggestion to add a kill switch for the idxstats step. Testing of the individual tool looks good and the full workflow should be done sometime tonight. Please review.

**Review by migbro — APPROVED** [review]
Adding the outputs descriptions were a nice touch. Not that important, but maybe a test with an input that has metadata to see if that inheritance function works would be nice.

**Review by mark-welsh — APPROVED** [review]
nice `awk` 😎

**Pull request merged**